src/java/org/apache/cassandra/service/accord/txn/TxnQuery.java

```diff
@@ -49,17 +54,47 @@ public Result compute(TxnId txnId, Data data, @Nullable Read read, @Nullable Upd
 
     public static final TxnQuery NONE = new TxnQuery()
     {
+        @Override
+        protected byte type()
+        {
+            return 2;
+        }
+
         @Override
         public Result compute(TxnId txnId, Data data, @Nullable Read read, @Nullable Update update)
         {
             return new TxnData();
         }
     };
 
+    public static final TxnQuery CONDITION = new TxnQuery()
+    {
+        @Override
+        protected byte type()
+        {
+            return 3;
+        }
+
+        @Override
+        public Result compute(TxnId txnId, Data data, @Nullable Read read, @Nullable Update update)
```

Comment:
`update` isn't nullable for CAS, so what happens on 82 is fine?

Reply:
Huh, yeah copy paste error from the others. Accord says that `Data` is also nullable in the interface.

I changed it and added null preconditions.